### PR TITLE
feat: add restriction for installing modules with forbidden mounted binaries [NET-428]

### PR DESCRIPTION
**Pull request by kmd-fl**
This PR restricts adding modules with mounted binaries not from the whitelist.

This check happens only in `ModuleRepository::add_module` function, which is called from `Dist.add_module/Dist.add_module_from_vault` and in the builtin-deployer.

The following binaries added in the whitelist by default: `/usr/bin/curl`, `/usr/bin/ipfs`.

To add binaries to the list via configuration file, you need to add to `Config.toml`:
```toml
allowed_binaries = [
   "/usr/bin/curl",
  "/usr/bin/ipfs"
] 
```
(~~I hope, it should work, didn't check yet, lol~~ checked)

### CODE REVIEW

File: crates/particle-node-tests/tests/services.rs [hunk 1]
```diff
@@ -21,12 +21,16 @@ use serde_json::Value as JValue;
...
 use connected_client::ConnectedClient;
-use created_swarm::make_swarms;
+use created_swarm::{make_swarms, make_swarms_with_cfg};
 use service_modules::load_module;
 
 #[tokio::test]
 async fn create_service_from_config() {
-    let swarms = make_swarms(1).await;
+    let swarms = make_swarms_with_cfg(1, |mut cfg| {
+        cfg.allowed_binaries = vec!["/tmp".to_string()];
```
[the remark this answers is not in the document]
Comment (kmd-fl) [reply crates/particle-node-tests/tests/services.rs:30]: It's a binary, I belive. If you look at the test, you see `/tmp` used as a binary.

**Comment by kmd-fl** [comment]
I thought about adding separate whitelists for management ids and other ids. For example, to allow ipfs only for aqua-ipfs service. But I think if we're gonna do this, it should be in a separate PR. Maybe it will be done when we have the full support of that effector thing.

File: particle-modules/src/modules.rs [hunk 2]
```diff
@@ -166,11 +170,29 @@ impl ModuleRepository {
 
         Ok(())
     }
+
+    fn check_module_mounted_binaries(&self, config: &TomlMarineNamedModuleConfig) -> Result<()> {
+        if let Some(binaries) = &config.config.mounted_binaries {
+            for requested_binary in binaries.values() {
+                if let Some(requested_binary) = requested_binary.as_str() {
+                    let requested_binary_path = Path::new(requested_binary).to_path_buf();
```
Comment (folex) [inline particle-modules/src/modules.rs:178]: why do you allocate here every time? you use reference anyway 🤔

**Comment by justprosh** [comment]
@folex because if you restrict all binaries now, you can't even start a peer, because aqua-ipfs can't be deployed

File: crates/particle-node-tests/tests/services.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (folex) [reply crates/particle-node-tests/tests/services.rs:30]: very strange... maybe change /tmp to something like use bin curl?

**Comment by kmd-fl** [comment]
@justprosh I added ipfs binary in the default whitelist.

File: particle-modules/src/modules.rs [hunk 3]
```diff
@@ -68,6 +69,7 @@ pub struct ModuleRepository {
     blueprints: Arc<RwLock<HashMap<String, Blueprint>>>,
     max_heap_size: ByteSize,
     default_heap_size: Option<ByteSize>,
+    allowed_binaries: Vec<PathBuf>,
```
Comment (folex) [inline particle-modules/src/modules.rs:72]: i think this should be a hashset

**Review by folex — APPROVED** [review]

**Pull request merged**